src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -369,9 +355,14 @@ class SourcesTree extends Component<Props, State> {
 }
 
 const mapStateToProps = state => {
+  const selectedSource = getSelectedSource(state);
+  const shownSource = getShownSource(state);
+
   return {
-    shownSource: getShownSource(state),
-    selectedSource: getSelectedSource(state),
+    shownSource,
+    shownSourceForTree: getSourceFromPrettySource(state, shownSource),
+    selectedSource,
+    selectedSourceForTree: getSourceFromPrettySource(state, selectedSource),
```

Comment:
I don't feel great about making these props.  I'd much rather keep this get-the-non-pretty-printed-source a utility function.  Thoughts @jasonLaster ?

Reply:
I like this as props/selectors for 2 reasons:

1. it simplifies the component
2. makes it easier to share / test the logic 

why would this be cleaner as a utility?